Cabal/Distribution/Lexer.x

```diff
@@ -0,0 +1,244 @@
+{
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Distribution.Parsec.Lexer
+-- License     :  BSD3
+--
+-- Maintainer  :  cabal-devel@haskell.org
+-- Portability :  portable
+--
+-- Lexer for the cabal files.
```

Comment:
Yeah, I wrote the initial version some years ago (alex lexer + parsec parser). Not really based on anything except the old parser plus lots of experiments with parsing all the existing hackage .cabal files.